lib/timezone/timezone.ex

```diff
@@ -77,6 +77,19 @@ defmodule Timex.Timezone do
   def name_of("Y"),    do: name_of(-12)
   def name_of("Z"),    do: "UTC"
   def name_of("UT"),   do: "UTC"
+  def name_of("+0330"), do: "Asia/Tehran"
```

Comment:
Yeah we definitely can't go this route, I would instead change this to return a variant of `Etc/GMT`, such as `Etc/GMT+5:45`, and modify `get` to special-case these and return a manually constructed `TimezoneInfo` struct.

For example, `Etc/GMT+5` from tzdata is:

```elixir
%{from: %{standard: :min, utc: :min, wall: :min}, std_off: 0,
    until: %{standard: :max, utc: :max, wall: :max}, utc_off: -18000,
    zone_abbr: "GMT+5"}
```

The same thing for `Etc/GMT+5:45` would be:

```elixir
%{from: %{standard: :min, utc: :min, wall: :min}, std_off: 0,
    until: %{standard: :max, utc: :max, wall: :max}, utc_off: -20700,
    zone_abbr: "GMT+5:45"}
```

And as a TimezoneInfo:

```elixir
"%TimezoneInfo{
  abbreviation: "GMT+5:45", 
  from: :min, 
  full_name: "Etc/GMT+5:45", 
  offset_std: 0, 
  offset_utc: -345, 
  until: :max}
```

This is the really the only valid way to parse these offsets, because we can't assume what zone was intended, only that the zone which was formatted had some offset - with date/times it's actually worse to assume more information than was given, and assuming the zone from an offset, even if a given offset is currently unique, is assuming too much.  At any time a new zone could be added, or an existing one changed, to use an odd offset like this. We can't ever make an assumption about what zone was meant (even if we can determine exactly what zone is referenced because it's the only one with such an offset at that time).

The alternative is to have some special handling in the timezone offset parsers which use the date/time context to see if it can locate the specific zone given the offset, but I dislike that because it's not consistent with how offsets have to be parsed in the general case (i.e. parsed into one of the etcetera zones, like `Etc/GMT+5`). Since the Olson timezone database does not contain any such etcetera zones, we have to roll it ourselves for cases like this.

Creating an etcetera zone for an odd offset is fine generally anyway when parsing datetimes and shifting them to another zone, since when shifting to another zone we only care about the current offset, not the rules of the zone, which only come into play when shifting through time - and combining parsing with shifts through time is a rare case, if done at all - and something I'd rather be able to say "this is how it behaves when you do that, for every offset", than to say "well sometimes we assume a zone based on the offset, and sometimes we just use the offset".

I think I've probably gone on too long, but wanted to make my justification clear :)